cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -196,11 +202,20 @@ public KafkaReconciler(
         this.operationTimeoutMs = config.getOperationTimeoutMs();
         this.kafkaNodePoolCrs = nodePools;
 
+        this.kafkaMetadataStateManager = new KafkaMetadataStateManager(reconciliation, kafkaCr);
+        KafkaMetadataStateManager.KafkaMetadataConfigurationState kafkaMetadataConfigState = this.kafkaMetadataStateManager.computeMetadataConfigurationState();
+        LOGGER.infoCr(reconciliation, "KafkaMetadataConfigurationState = {}", kafkaMetadataConfigState);
+
+        // TODO: to evaluate this one, because used only for inter broker protocol version and log message format version below
         boolean isKRaftEnabled = config.featureGates().useKRaftEnabled() && ReconcilerUtils.kraftEnabled(kafkaCr);
+        // KRaft to be considered not only when fully enabled (KRAFT = 4) but also when a migration is about to start (PRE_MIGRATION = 1)
+        // NOTE: this is important to drive the right validation happening in node pools (i.e. roles on node pools, storage, number of controllers, ...)
+        boolean isKRaftInConfiguration = config.featureGates().useKRaftEnabled() &&
```

Comment:
`isKRaftInConfiguration` means what? It sounds like whether user configured something. I think the name is confusing.

Reply:
Compared to the old code, it's not feature gate enabled AND cluster is full KRaft anymore. 
Now you could have KRaft configured in a node but the cluster is not full KRaft yet because there is a migration going on (i.e. from the PRE_MIGRATION state up to KRAFT). This is the point I am trying to make with this name ... something like "KRaft IS in the configuration but cluster is NOT necessarily KRaft yet".
It means that even in this case the node pools creation stuff for example has to run some validation (i.e. warning about having just 2 controllers, when you create them to migrate).
Do you have any better suggestion?

Reply:
That line was removed and simplified.